src/axom/inlet/Inlet.cpp

```diff
@@ -184,10 +190,77 @@ void Inlet::registerDocWriter(std::shared_ptr<DocWriter> writer) {
 }
 
 void Inlet::writeDoc() {
-  if (m_docWriterEnabled) {
+  if (m_docEnabled) {
     m_docWriter->writeDocumentation();
   }
 }
 
+bool Inlet::verify() {
+  bool verifySuccess = true;
+  verifyRecursive(m_sidreRootGroup, verifySuccess);
+  return verifySuccess;
+}
+
+void Inlet::verifyRecursive(axom::sidre::Group* sidreGroup, bool& verifySuccess) {
+  SLIC_ASSERT_MSG(sidreGroup, "Root was nullptr");
+  if (sidreGroup == m_sidreRootGroup && sidreGroup->hasView("warningFlag")) {
+    verifySuccess = false;
+  }
+
+  if (sidreGroup->hasView("required")) {
+    int8 required = sidreGroup->getView("required")->getData();
+    if (required && !sidreGroup->hasView("value")) {
+      std::string msg = fmt::format("Inlet: {0}: Required field was not specified in Input Deck", 
+                                    sidreGroup->getPathName());
+      SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+      verifySuccess = false;
+    }
+  }
+
+  if (sidreGroup->hasView("value")) {
+    if (sidreGroup->hasView("range")) {
+      auto type = sidreGroup->getView("range")->getTypeID();
+      if (type == axom::sidre::INT_ID) {
+        int* range = sidreGroup->getView("range")->getArray();
+        int val = sidreGroup->getView("value")->getScalar();
+        // Checks if the value is not within the specified range
+        if (!(range[0] <= val && val <= range[1])) {
+          verifySuccess = false;
+        }
+      } else if (type == axom::sidre::DOUBLE_ID) {
+        double* range = sidreGroup->getView("range")->getArray();
+        double val = sidreGroup->getView("value")->getScalar();
+        if (!(range[0] <= val && val <= range[1])) {
+          verifySuccess = false;
+        }
+      } else {
+        verifySuccess = false;
+      }
+    } else if (sidreGroup->hasView("validValues")) {
+      int val = sidreGroup->getView("value")->getScalar();
+      int* range = sidreGroup->getView("validValues")->getArray();
+      size_t size = sidreGroup->getView("validValues")->getBuffer()->getNumElements();
+      // Checks if the value specified is one of the valid values
+      bool found = false;
```

Comment:
Possible optimization: If you sort the array, you can use ``std::binary_search``
(although, it is not clear that this would matter in practice)

Reply:
Probably a good idea to hold off on this until the need arises

Reply:
On second thought, this is probably overkill for this use-case and can be skipped for now.